### ISSUE TITLE
Unable to use shouldAdvanceTime and advanceTimeDelta options.

### ISSUE THREAD

**Issue by royalpinto** [sinonjs/lolex#122]
Env Details:
`lolex: 2.1.0`
`node: v6.9.2`

Following script throws an error:
```js
var lolex = require('lolex');
var clock = lolex.install({shouldAdvanceTime: true, advanceTimeDelta: 40});
setTimeout(() => console.log(1), 4000);
```

Error:
```bash
1
_stream_writable.js:396
  if (state.length === 0 && state.needDrain) {
           ^

TypeError: Cannot read property 'length' of undefined
    at onwriteDrain (_stream_writable.js:396:12)
    at afterWrite (_stream_writable.js:386:5)
    at runJobs (/Users/lohith.pinto/Development/stark/node_modules/lolex/src/lolex-src.js:183:18)
    at Object.tick (/Users/lohith.pinto/Development/stark/node_modules/lolex/src/lolex-src.js:550:9)
    at doIntervalTick (/Users/lohith.pinto/Development/stark/node_modules/lolex/src/lolex-src.js:411:11)
    at ontimeout (timers.js:365:14)
    at tryOnTimeout (timers.js:237:5)
    at Timer.listOnTimeout (timers.js:207:5)
```

What could be the issue ?

**Comment by fatso83** [comment]
Thanks for your bug report. This was quite unfortunate.

@elad-nach, any ideas?

**Comment by elad-nach** [comment]
@fatso83 I'm just looking into this as I'm trying to integrate 2.1.0 into sinon.

I'm also getting sporadic errors there - regardless to whether lolex is installed with `shouldAdvanceTime = true`. Errors are shown on mocha in the XHR test section on sinon.
I'm investigating in any case. will update once I nail it.

**Comment by elad-nach** [comment]
@fatso83 @benjamingr 
could you please have a second look at the `nextTick()` feature that was merged? The `runJobs` was added on that PR.
see [here](https://github.com/angular/zone.js/issues/760) and [here](https://github.com/angular/zone.js/pull/762/files).

**Comment by benjamingr** [comment]
The error makes sense - the user overrode `process.nextTick` and did not uninstall the timer - Node was trying to flush the ticks and lolex was there.

I'll see if I can hack around it (by mimicing nextTick better), the obvious fix is to advance the clock synchronously and then uninstall. 

A general way to reproduce would be:

```js
lolex.install({}); // 
startSomeAsyncNodeAction((err, data) => {
 // startSomeAsyncNodeAction might call nextTick
});
```

**Comment by benjamingr** [comment]
On it.

**Comment by benjamingr** [comment]
I think I know why this might be - we change how timers work _after_ the first tick in Node, @royalpinto can you try to reproduce with:

```js
var lolex = require('lolex');
var clock = lolex.install({shouldAdvanceTime: true, advanceTimeDelta: 40});
setTimeout(() => console.log(1), 4000);
```

**Comment by benjamingr** [comment]
It's because `nextTick` can take extra parameters, but we never documented it - I'll fix.

**Comment by benjamingr** [comment]
Sorry, this is my bad - I _meant_ to have it work but I made a silly mistake:

```
-            args: Array.prototype.slice.call(1)
+            args: Array.prototype.slice.call(arguments, 1)
```

Submitting a PR now

**Comment by royalpinto** [comment]
@fatso83 @elad-nach @benjamingr 
Thank you all for the quick resolution 👍 
Is this is available in the new `npm` release now ? I did a `npm install lolex` (it installed same `2.1.0`) and error is still repeating.

**Comment by fatso83** [comment]
@royalpinto  [No new patch release has been made](https://github.com/sinonjs/lolex/blob/master/History.md). I'll see to it today.

**Comment by fatso83** [comment]
This is now published as [2.1.1](https://www.npmjs.com/package/lolex).
